pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -7,18 +7,75 @@
 # include <string.h>
 # include <stdarg.h>
 # include <stdlib.h>
+# include <stdio.h>
+
+/*
+ * Takes an array, and prints its elements the way they are laid out in memory (similar to ravel)
+*/
+
+void print_ndarray_memory(t_ndarray nd)
+{
+    int i = 0;
+    while (i < nd.length)
+    {
+        switch (nd.type)
+        {
+            case nd_int8:
+                printf("[%hhd]", nd.nd_int8[i]);
+                break;
+            case nd_int16:
+                printf("[%hd]", nd.nd_int16[i]);
+                break;
+            case nd_int32:
+                printf("[%d]", nd.nd_int32[i]);
+                break;
+            case nd_int64:
+                printf("[%ld]", nd.nd_int64[i]);
+                break;
+            case nd_float:
+                printf("[%f]", nd.nd_float[i]);
+                break;
+            case nd_double:
+                printf("[%lf]", nd.nd_double[i]);
+                break;
+            case nd_bool:
+                printf("[%d]", nd.nd_bool[i]);
+                break;
+            case nd_cfloat:
+            {
+                double real = creal(nd.nd_cfloat[i]);
+                double imag = cimag(nd.nd_cfloat[i]);
+                printf("[%lf%s%lfj]", real, imag >= 0 ? "+" : "", imag);
+                break;
+            }
+            case nd_cdouble:
+            {
+                double real = creal(nd.nd_cdouble[i]);
+                double imag = cimag(nd.nd_cdouble[i]);
+                printf("[%lf%s%lfj]", real, imag >= 0 ? "+" : "", imag);
+                break;
+            }
+            default:
+                return;

```

Comment:
Why is return the default?

Reply:
I tought it should do nothing if the type is unknown, it should also break from the loop

Reply:
These are homogeneous right? In that case I understand why you return but it is strange that no error is raised you just print `""`. Personally I would probably have not given a default, then you should get a compiler warning if a new object is added to the enum

Reply:
An error does make sense

I didn't understand what you mean
> then you should get a compiler warning if a new object is added to the enum

Reply:
`nd_bool` etc are members of an enum aren't they?

If I compile:
```c
#include <stdio.h>

int main(int argc, char** argv)
{
    enum my_enum {
        TagA, TagB
    };

    enum my_enum a =TagA;

    switch(a) {
        case TagA: printf("A\n"); break;
    };
}
```
with `gcc -Wall -c tmp.c` then I get:
```c
tmp.c: In function ‘main’:
tmp.c:11:5: warning: enumeration value ‘TagB’ not handled in switch [-Wswitch]
   11 |     switch(a) {
      |     ^~~~~~
```
So it is obvious when something hasn't been implemented. If I use `default` then the warning is suppressed and it is easier for me to forget about this if I add more options to my enum later.

Of course this assumes that you already handle all the tags in the enum currently

Reply:
https://github.com/pyccel/pyccel/pull/1128/commits/c061f83fb9860c823980c3bb90039d78d368c480